codebuild/bin/s2n_open_fds_test.py

```diff
@@ -0,0 +1,81 @@
+import os
+import re
+import sys
+
+# Allow stdin, stdout, stderr, and the file descriptor for dynamic analysis log to remain open at exit
+ACCEPTABLE_OPEN_FDS = 4
+ERROR_EXIT_CODE = 123
+
+analysis_file_location = "../../build/Testing/Temporary"
+analysis_file_pattern = re.compile(r"^LastDynamicAnalysis*")
+# This regular expression captures valgrind 3.13 and valgrind 3.18+ log
```

Comment:
> Do we need to test for invalid versions or an invalid log pattern? 

No, we don't.

> Should the script handle invalid logs gracefully, or are we only using these versions?

I don't think there are any logs that are considered invalid.

The purpose of this comment is to say that Valgrind 3.13 in Ubuntu 18's error messages is slightly different from those in Ubuntu 22 and 24. Hence, I wrote a slightly more generic regex to capture error messages from different Valgrinds.

Output from Valgrind 3.13 in Ubuntu 18:
```
FILE DESCRIPTORS: 5 open at exit.
```

Output from Valgrind 3.18+ in Ubuntu 22 and 24:
```
FILE DESCRIPTORS: 5 open (3 std) at exit.
```


I made that `(3 std)` optional, so that the script can parse error messages from all Valgrinds.

Reply:
Discussed offline.